stable/jenkins/templates/jenkins-master-networkpolicy.yaml

```diff
@@ -0,0 +1,33 @@
+{{- if .Values.NetworkPolicy.Enabled }}
+kind: NetworkPolicy
+apiVersion: {{ .NetworkPolicy.ApiVersion }}
+metadata:
+  name: "{{.Release.Name}}-{{.Values.Master.Component}}"
+spec:
+  podSelector:
+    matchLabels:
+      component: "{{.Release.Name}}-{{.Values.Master.Component}}"
+  ingress:
+    # Allow web access to the UI
+    - ports:
+      - port: {{.Values.Master.ContainerPort}}
+    # Allow inbound connections from slave
+    - from:
+      - podSelector:
+          matchLabels:
+            "jenkins/{{.Release.Name}}-{{.Values.Agent.Component}}": "true"
```

Comment:
Jenkins agents are created by the jnlp-agent plugin. The jnlp plugin creates labels of the following form: `jenkins/<label>: "true"` where `<label>` is the value set in the [jnlp config](https://github.com/kubernetes/charts/pull/1228/files#diff-7146a5152507fba3404ff5118aab029cR38). 

For example, when we specify `<label>squeaky-penguin-jenkins-slave</label>` in our jnlp config, the assigns the label `jenkins/squeaky-penguin-slave: "true"` for the agent pod.

For this chart, we must conform to the label scheme used by jnlp-agent plugin. For other charts, we can come up with our own schema.